doc/services/logging/index.rst

```diff
@@ -566,6 +565,130 @@ Logging v2
 String arguments are handled by :ref:`cbprintf_packaging` thus no special action
 is required.
 
+Multi-domain support
+====================
+
+More complex systems may consist of multiple domains where domain is an
+independent binary. An example of a domain is a core in the multicore SoC or one
+of the binaries on ARM TrustZone core (Secure and Nonsecure). Tracing and
+debugging on multidomain system is more complex and requires efficent logging
+system. One option is to log in each domain independently. However, this option
+is not always possible since it requires that each domain has available backend
+(e.g. UART). Additionally, in that approach logs are presented on independent
+outputs which is troublesome to use and not scalable. Another option is to use
+multi-domain logging where log messages from each domain end up in one root domain
+where they are processed in the same way as in a single domain case.
+
+Log messages are passed between domains using a connection between domains which
+is created from backend on one side and link on the other. Log link is an
+interface introduced in multi-domain approach. Log link is responsible for
+receiving log message from another domain, create a copy and put that local
+log message (but with remote data) into the message queue. Specific log link
+implementation matches complementary backend implementation to allow log
+messages exchange and logger control like configuring filtering, getting log
+source names, etc.
+
+There are 3 types of domains in the multi-domain system:
+
+- ``End domain`` which has the logging core implementation and cross domain
+  backend. It can also have other backends in parallel.
+- ``Relay domain`` which has one or more links to other domains but does not
+  have backend(s) which outputs logs to the user. It has cross domain backend to
+  another relay or root domain.
+- ``Root domain`` which has one or multiple links and backend which outputs logs
+  to the user.
+
+Example of multi-domain setup is shown below.
+
+.. _logging-multidomain-example:
+
+.. figure:: images/multidomain.png
+
+    Multi-domain example
```

Comment:
```suggestion
There are three types of domains in the multi-domain system:

* *End domain*, which has the logging core implementation and a cross-domain
  backend. It can also have other backends in parallel.
* *Relay domain*, which has one or more links to other domains but does not
  have backends that output logs to the user. It has a cross-domain backend either to
  another relay or to the root domain.
* *Root domain*, which has one or multiple links and a backend that outputs logs
  to the user.

See the following image for an example of a multi-domain setup:

.. _logging-multidomain-example:

.. figure:: images/multidomain.png

    Multi-domain example
```